src/sentry/web/api.py

```diff
@@ -112,6 +117,36 @@ def _parse_header(self, request, helper, project):
 
         return auth
 
+    def _publish_to_kafka(self, request):
+        """
+        Sends raw event data to Kafka for later offline processing.
+        """
+        try:
+            if len(request.body) > options.get('kafka-publisher.max-event-size'):
+                return
+
+            # Sampling
+            if random.random() >= options.get('kafka-publisher.raw-event-sample-rate'):
+                return
+
+            # We want to send only serializable items from request.META
+            meta = {}
+            for key, value in request.META.items():
+                try:
+                    json.dumps([key, value])
+                    meta[key] = value
+                except TypeError:
+                    pass
+
+            meta['SENTRY_API_VIEW_NAME'] = self.__class__.__name__
+
+            kafka_publisher.publish(
+                channel=getattr(settings, 'KAFKA_RAW_EVENTS_PUBLISHER_TOPIC', 'raw_store_events'),
+                value=json.dumps([meta, request.body])
+            )
+        except Exception as e:
+            logger.debug("Cannot publish event to Kafka: {}".format(e.message))
```

Comment:
Do we have debug logging on in production? Or is this intended as a development aid only?

Reply:
Mainly for development/debugging. I didn't want to make it `logger.exception` or anything like that because then it _theoritically_ can be triggered for every event.